model/Export/ExportForm.php

```diff
@@ -56,108 +57,146 @@ abstract class ExportForm extends tao_helpers_form_FormContainer
-                            $this->isInstanceValid($instance);
-                        } catch (\common_exception_UserReadableException $e) {
-                            $disabledOptions[$instance->getUri()] = $e->getUserMessage();
-                        }
-                        $options[$instance->getUri()] = $instance->getLabel();
-                    }
-                }
+                $fileName = $this->getFileName($class->getLabel());
+                $options = $this->getOptions(...$class->getInstances(true));
             }
         }
 
         $nameElt = tao_helpers_form_FormFactory::getElement('filename', 'Textbox');
         $nameElt->setDescription(__('File name'));
         $nameElt->setValue($fileName);
-        $nameElt->setUnit(".zip");
+        $nameElt->setUnit('.zip');
         $nameElt->addValidator(tao_helpers_form_FormFactory::getValidator('NotEmpty'));
         $this->form->addElement($nameElt);
 
         $instanceElt = tao_helpers_form_FormFactory::getElement('instances', 'Checkbox');
         $instanceElt->setDescription(__('Items'));
 
-        if (count($options) > 1) {
+        if (count($options['options']) > 1) {
             $instanceElt->setAttribute('checkAll', true);
         }
 
-        $instanceElt->setOptions(tao_helpers_Uri::encodeArray($options, tao_helpers_Uri::ENCODE_ARRAY_KEYS));
-        $instanceElt->setReadOnly(tao_helpers_Uri::encodeArray($disabledOptions, tao_helpers_Uri::ENCODE_ARRAY_KEYS));
-        foreach (array_keys($options) as $value) {
-            if (!isset($disabledOptions[$value])) {
+        $instanceElt->setOptions(tao_helpers_Uri::encodeArray($options['options'], tao_helpers_Uri::ENCODE_ARRAY_KEYS));
+        $instanceElt->setReadOnly(
+            tao_helpers_Uri::encodeArray($options['disabledOptions'], tao_helpers_Uri::ENCODE_ARRAY_KEYS)
+        );
+        foreach (array_keys($options['options']) as $value) {
+            if (!isset($options['disabledOptions'][$value])) {
                 $instanceElt->setValue($value);
             }
         }
         $this->form->addElement($instanceElt);
 
-
         $this->form->createGroup('options', '<h3>' . $this->getFormGroupName() . '</h3>', ['filename', 'instances']);
     }
 
-    private function isInstanceValid($item)
+    /**
+     * @return taoItems_models_classes_ItemsService
+     */
+    protected function getItemsService()
+    {
+        return taoItems_models_classes_ItemsService::singleton();
+    }
+
+    protected function getFileName(string $name): string
     {
+        return strtolower(tao_helpers_Display::textCleaner($name, '*'));
+    }
+
+    protected function getOptions(core_kernel_classes_Resource ...$resource): array
+    {
+        $itemService = taoItems_models_classes_ItemsService::singleton();
+        $options = [];
+        $disabledOptions = [];
+        foreach ($resource as $instance) {
+            if ($itemService->hasItemModel($instance, [ItemModel::MODEL_URI])) {
```

Comment:
Early return is preferable here IMO, too much of nesting is happening here
Consider doing `if (!$itemService->hasItemModel($instance, [ItemModel::MODEL_URI])) break;`

Reply:
`continue?` or i don't see something?

Reply:
Right, `continue`